### PR TITLE
add regex for US account number

### PR DESCRIPTION
**Pull request by thesahindia**
### Details
Added regex for US account number

### Fixed Issues

$ https://github.com/Expensify/App/issues/7056

### Tests

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Open app
2. Navigate to settings > Workspace > Add bank account
3. Proceed to the manual flow
4. Add alphabets in account number
- [x] Verify that no errors appear in the JS console
- [x] Verify that account number shows error for alphanumeric characters.
- [x] Verify that account number works for numbers of length 4-17

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web

https://user-images.githubusercontent.com/83179295/148772918-15e7edea-e261-4bc2-b92a-2bdef153737a.mov

### CODE REVIEW

**Comment by rushatgabhane** [comment]
Hey @thesahindia, could you please add QA steps something like - 
Verify that account number shows error for alphanumeric characters.

Verify that account number works for numbers of length 4-17

**Comment by rushatgabhane** [comment]
Logs show there's some error with bot's GPG key.
I wouldn't worry about it. You can run tests locally too btw.

File: src/CONST.js [hunk 1]
```diff
@@ -72,6 +72,7 @@ const CONST = {
         },
         REGEX: {
             IBAN: /^[A-Za-z0-9]{2,30}$/,
```
Comment (rushatgabhane) [inline src/CONST.js:74]: IBAN regex isn't used anywhere. We're safe to remove it.

**Review by rushatgabhane — COMMENTED** [review]
Looks good. Suggested some changes.

File: src/CONST.js (hunk 1 above)
Comment (thesahindia) [reply src/CONST.js:74]: Yeah we can remove it.

**Review by rushatgabhane — DISMISSED** [review]
LGTM! 🎉
All yours @puneetlath

File: src/pages/ReimbursementAccount/BankAccountStep.js [hunk 2]
```diff
@@ -88,7 +88,7 @@ class BankAccountStep extends React.Component {
         const errors = {};
 
         // These are taken from BankCountry.js in Web-Secure
```
Comment (puneetlath) [inline src/pages/ReimbursementAccount/BankAccountStep.js:90]: We should remove this comment as it is no longer relevant.

**Review by puneetlath — CHANGES_REQUESTED** [review]
One small comment. Otherwise looks good.

**Comment by puneetlath** [comment]
Looks good @thesahindia. Can you please merge main to fix the failing tests? Context here: https://expensify.slack.com/archives/C01GTK53T8Q/p1641911028289600

**Review by puneetlath — APPROVED** [review]

**Pull request merged**